src/site/content/en/blog/rendering-on-the-web/index.md

```diff
@@ -2,349 +2,169 @@
+
+Experiences that use client-side rendering that rely on large JavaScript bundles should consider [aggressive code-splitting](/reduce-javascript-payloads-with-code-splitting/) to lower TBT and INP during page load, and be sure to lazy-load JavaScript—"serve only what you need, when you need it". For experiences with little or no interactivity, server rendering can represent a more scalable solution to these issues.
+
+For folks building single page applications, identifying core parts of the user interface shared by most pages means you can apply the [application shell caching](https://developer.chrome.com/blog/app-shell/) technique. Combined with service workers, this can dramatically improve perceived performance on repeat visits, as the application shell HTML and its dependencies can be loaded from `CacheStorage` very quickly.
+
+## Combining server rendering and client-side rendering via rehydration
+
+This approach attempts to smooth over the trade-offs between client-side rendering and server rendering by doing both. Navigation requests like full page loads or reloads are handled by a server that renders the application to HTML, then the JavaScript and data used for rendering is embedded into the resulting document. When done carefully, this achieves a fast FCP just like server rendering, then "picks up" by rendering again on the client using a technique called [(re)hydration](https://react.dev/reference/react-dom/client/hydrateRoot). This is an effective solution, but it can come with considerable performance drawbacks.
+
+The primary downside of server-side rendering with rehydration is that it can have a significant negative impact on TBT and INP, even if it improves FCP. Server-side rendered pages can deceptively appear to be loaded and interactive, but can't actually respond to input until the client-side scripts for components are executed and event handlers have been attached. This can take seconds or even minutes on mobile.
+
+Perhaps you've experienced this yourself—for a period of time after it looks like a page has loaded, clicking or tapping does nothing. This quickly becoming frustrating, as the user is left to wonder why nothing is happening when they try to interact with the page.
+
+### A rehydration problem: one app for the price of two
+
+Rehydration issues can often be worse than delayed interactivity due to JavaScript. In order for the client-side JavaScript to be able to accurately "pick up" where the server left off without having to re-request all of the data the server used to render its HTML, current server-side rendering solutions generally serialize the response from a UI's data dependencies into the document as script tags. The resulting HTML document contains a high level of duplication:
```

Comment:
I'm really confused between the difference between "prerendering on server" and rehydration. Are they not the same thing?